.github/workflows/build.yml

```diff
@@ -20,11 +20,10 @@ jobs:
                   java-version: 17
                   cache: maven
             - name: Cache SonarCloud packages
-              uses: actions/cache@v1
+              uses: actions/cache@v4
               with:
                   path: ~/.sonar/cache
                   key: ${{ runner.os }}-sonar
-                  restore-keys: ${{ runner.os }}-sonar
```

Comment:
Not clear why you are proposing to remove this line? Do you have a reason?

Reply:
I found the `key` and `restore-keys` are the same, and we could omit the `restore-keys` config according to <https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#using-the-cache-action>.